src/migrateContentServer.ts

```diff
@@ -0,0 +1,181 @@
+import _fromPairs from "lodash/fromPairs";
+import _mapValues from "lodash/mapValues";
+import { ContentRecord } from "@activeviam/activeui-sdk-5.1";
+import { getIndexedDataModel, DataModel } from "@activeviam/data-model-5.1";
+import {
+  BehaviorOnError,
+  OutcomeCounters,
+  MigrationFunction,
+} from "./migration.types";
+import { getMigrateDashboards } from "./getMigrateDashboards";
+import { getMigrateSavedWidgets } from "./getMigrateSavedWidgets";
+import { getMigrateSavedFilters } from "./getMigrateSavedFilters";
+import { migrate_43_to_50 } from "./4.3_to_5.0";
+import { migrate_50_to_51 } from "./5.0_to_5.1";
+import { getContent } from "./getContent";
+import {
+  AtotiUIFromVersion,
+  AtotiUIToVersion,
+} from "./convertAtotiToAUIVersions";
+
+const migrationSteps: {
+  from: string;
+  to: string;
+  migrate: MigrationFunction;
+}[] = [{ from: "5.0", to: "5.1", migrate: migrate_50_to_51 }];
+
+const fromVersions = migrationSteps.map(({ from }) => from);
+const toVersions = migrationSteps.map(({ to }) => to);
+
+/**
+ * Migrates `contentServer` from an older version of Atoti UI to a more recent one.
+ * Also keeps track of the number of migration successes and failures in `counters` and a detailed `errorReport`.
+ *
+ * Mutates `contentServer`, `errorReport` and `counters`.
+ */
+export async function migrateContentServer({
```

Comment:
`migrateContentServer` vs `migrateContentServerJson` does not seem confusing to me.
And it has the pro of keeping: one CLI command = one script (without extra work on top of it).
That being said, it's not a strong opinion, and I like the idea of encapsulating the logging of the outcome anyway 👍 

To be consistent I did the same for `migrateNotebook`.
While doing this, I notice that `migrateNotebook` is not really generic contrarily to `migrateContentServer`.
We'd have to adapt it in order to make it work with several iterative steps.
Also, the outcome logging/error reporting should be the same IMO, except that it should point to notebook cells rather than content server files.
This is out of scope of the PR.

----

The current design was the outcome of a technical design session involving us two, Thibault and Elias.
I'm not sure if there is a more complete design document, but here is a [Slack thread](https://activeviam.slack.com/archives/C04BW2ALHL2/p1669050176710479) which shows us agreeing on this almost exact function split/naming. 

If I remember correctly, our goal was to make it as easy as possible to write the actual migration logic, as opposed to worrying about the manipulation of entries on the content server and having to duplicate a lot of code/reinvent the wheel for each migration step.

And so we were ready to have some generic abstraction/indirections that are not the easiest to read, if it allowed to achieve this goal.
Which I think we did: as for example it's quite easy to write one `migrateWidget` function per migration step, and the structure is consistent.

I'm not married at all to the current design and would be happy to discuss improvements, but I think it's nice to remember our initial intent if we do go for a redesign.

Reply:
I remember this session indeed. I've just felt like I needed way more time than I should in order to get productive here in the recent work I've had to do here and it's tempting to look at refactoring without removing the isolution of the pure widgetState migration.